metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/StudySteps.java

```diff
@@ -0,0 +1,138 @@
+package uk.ac.ebi.ampt2d.metadata.cucumber;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import cucumber.api.java.en.When;
+import org.junit.Ignore;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Ignore
+@AutoConfigureMockMvc
+public class StudySteps {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+
+    @When("user create a test study with (.*) for taxonomy$")
+    public void createTestStudy(String testTaxonomyKey) throws Exception {
+        CommonStates.setResultActions(postTestStudy("EGAS0001", 1, "test_human_study", false, LocalDate.now(), testTaxonomyKey));
+    }
+
+    @When("user create a test parameterized study with (.*) for accession, (.*) for version, (.*) for name, (.*) for deprecated, (.*) for releaseDay and (.*) for taxonomy$")
+    public void createTestStudyParameterizedMore(String accession, int version, String name, boolean deprecated, int releaseDay, String testTaxonomyKey) throws Exception {
+        CommonStates.setResultActions(postTestStudy(accession, version, name, deprecated, LocalDate.now().plusDays(releaseDay), testTaxonomyKey));
+    }
+
+    @When("^user request GET for the (.*)")
+    public void performGetOnResources(String className) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get("/" + className)));
+    }
+
+    @When("^user request GET for (.*) with query param (.*)")
+    public void performGetOnResourcesQuery(String className, String param) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get("/" + className + "?" + param)));
+    }
+
+    @When("^user request POST with (.*) for Uri (.*) for stringData (.*) for linkedObjectKey and (.*) for linkedObjectClassName")
```

Comment:
It looks like this step is used only for creating a new `Taxonomy`. Could you make this more specific and plain? Bear in mind that one point of Cucumber test is for non software developers to write the scenarios.